components/apimgt/org.wso2.carbon.apimgt.rest.api.webserver/src/main/java/org/wso2/carbon/apimgt/rest/api/webserver/impl/PublisherApiServiceImpl.java

```diff
@@ -48,7 +48,7 @@
             //#TODO read from config file
         /* TODO: Check the dot containment in last segment separated by '/' or use regex to capture file extension */
             if (app.equals(context)) {
-                if (rawUri.contains(".")) {
+                if (rawUri.split("\\?")[0].contains(".")) {
                     filePath = "./deployment/webapps" + path;
```

Comment:
Move "./deployment/webapps" to a string constant.
Can't we remove the TODO comment now?